src/ucs/sys/string.c

```diff
@@ -276,35 +276,50 @@ char *ucs_strtrim(char *str)
     return start;
 }
 
-const char * ucs_str_dump_hex(const void* data, size_t length, char *buf,
+const char *ucs_str_dump_hex(const void* data, size_t length, char *buf,
                               size_t max, size_t per_line)
 {
-    static const char hexchars[] = "0123456789abcdef";
+    return ucs_str_dump_hex_bulk(data, length, buf, max, per_line, 4, NULL);
+}
+
+const char *ucs_str_dump_hex_bulk(const void* data, size_t length, char *buf,
+                                  size_t max, size_t per_line, size_t word_size,
+                                  const char* prefix)
+{
+    size_t prefix_len = 0;
     char *p, *endp;
     uint8_t value;
     size_t i;
 
     p     = buf;
     endp  = buf + max - 2;
     i     = 0;
+
+    if (prefix != NULL) {
```

Comment:
can we avoid supporting prefix==NULL to simplify?
user that needs prefix can pass it including ": "